Platform/ElkhartlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -1884,7 +1884,27 @@ UpdateOsBootMediumInfo (
   OUT  OS_BOOT_OPTION_LIST  *OsBootOptionList
 )
 {
-  FillBootOptionListFromCfgData (OsBootOptionList);
+  OS_BOOT_OPTION             *BootOptionCfgData;
+  UINT32                      Idx;
+
+  for (Idx = 0; Idx < MAX_BOOT_OPTION_CFGDATA_ENTRY; Idx++) {
+    BootOptionCfgData = (OS_BOOT_OPTION *)FindConfigDataByTag (CDATA_BOOT_OPTION_TAG + Idx);
+    if (BootOptionCfgData->DevType == 0x5){
+      if (FeaturePcdGet (PcdPreOsCheckerEnabled) && PchIsSciSupported ()) {;
+        continue;
+      }
+      else{
+        BootOptionCfgData->BootFlags = 0x10;
+      }
+    }
+    if (BootOptionCfgData->ImageType != 0x00){
+      if (FeaturePcdGet (PcdPreOsCheckerEnabled) && PchIsSciSupported ()) {
+        FillBootOptionListFromCfgData (OsBootOptionList);
+      }
+    }else{
+      FillBootOptionListFromCfgData (OsBootOptionList);

```

Comment:
isn't this the same action as the #if -> FillBootOptionListFromCfgData (OsBootOptionList) ?